sklego/preprocessing.py

```diff
@@ -174,6 +174,10 @@ class ColumnSelector(BaseEstimator, TransformerMixin):
     """
 
     def __init__(self, columns: list):
+        # if the columns parameter is not a list, make it into a list
+        if not isinstance(columns, list):
+            columns = [columns]
+
```

Comment:
The following if statement is moved from fit to init method. This is done for two reasons first to make the code consisent with other similar classes(in this case ColumnDroper class). And second because this checking is suitable during class instantiation and has nothing specific to fit method.

Reply:
Normally I would agree but this isn't how scikit-learn works. When an estimator is cloned the attributes are set directly, so this overrides whatever happens in the `__init__`. Therefore the convention is to do all parameter validation and modification inside the fit method

Reply:
> Normally I would agree but this isn't how scikit-learn works. When an estimator is cloned the attributes are set directly, so this overrides whatever happens in the `__init__`. Therefore the convention is to do all parameter validation and modification inside the fit method

In that case we need to do the same thing in ColumnDroper class so to make the code consistent otherwise readers get confused why the same thing is done in 2 different ways (as I got when I first saw it), this may miss-lead some of them.

In either case we should change one of the class and make the code same for same functionality

Reply:
Agreed, if `ColumnDropper` does it the other way around atm that is a bug.